### PR TITLE
[Improvement-3767][api] Task instance supports query by process instance name

### PR DESCRIPTION
**Pull request by zhuangchong**
## What is the purpose of the pull request

#3767 

Task instances support query by process instance name. At present, when there are many instances on the task instance page, it is inconvenient to find a certain process instance. After improvement, you can query according to the process instance name on the task instance page.

## Brief change log

TaskInstance/index.vue adds workflow instance name query box
TaskInstanceController.queryTaskListPaging() added new input parameter processInstanceName

## Verify this pull request
This change added tests and can be verified as follows:
  - *Manually verified the change by testing locally.*

---
##拉取请求的目的是什么

任务实例支持按工作流实例名查询，目前在任务实例页面存在很多实例的时候，查找某一个工作流实例很不方便，改进后可以在任务实例页面根据工作流实例名查询

##简要更改日志

TaskInstance/index.vue 增加 工作流实例名称查询框
TaskInstanceController.queryTaskListPaging()新增入参processInstanceName

##验证此拉取请求

此更改增加了测试，可以进行如下验证：
   -*通过本地测试手动验证更改。*

image

### CODE REVIEW

**Comment by yangyichao-mango** [comment]
Please resolve the sonar duplication check.

**Comment by zhuangchong** [comment]
@break60  sonar check 8.7% Duplication，It’s because of the front-end common.js. Common.js just moved from instanceConditions to the conditions/instance directory. According to what you said, creating a new pr and submitting common.js related code did not solve this problem. Please preview

---
sonar 检查 8.7% 代码重复，是因为前端common.js 的原因，common.js 只是从instanceConditions 移动到conditions/instance目录下，按照您说的，新建pr提交common.js相关代码，并没有解决这个问题，帮preview一下

**Review by break60 — APPROVED** [review]
front end +1

**Comment by zhuangchong** [comment]
@qiaozhanwei @lgcareer done. Please help to review.

**Review by CalvinKirs — COMMENTED** [review]
The overall code seems to be no problem, but the only possible problem is to like to query this , Let's listen to other people's opinions

**Comment by zhuangchong** [comment]
When you want to find the task instance of the multi-task node corresponding to the current workflow instance, you now need to search by time or task name. However, when there are many task instances and a workflow definition corresponds to a multi-node task, the search is quite inconvenient. By adding the name query of the workflow instance, it will be convenient to query the task instance corresponding to the current workflow instance, and to view the task log and task information.

---

想要找当前工作流实例对应的多任务节点的任务实例时，现在需要按时间或任务名查找，但是任务实例很多并且一个工作流定义对应多节点任务的时候，查找就相当不便，本次新增通过工作流实例名查询，会很方便查询当前工作流实例对应的任务实例，进行查看任务日志和任务信息。

**Review by dailidong — APPROVED** [review]
+1
good job

**Review by CalvinKirs — APPROVED** [review]
+1

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TaskInstanceController.java [hunk 1]
```diff
@@ -95,11 +110,20 @@ public Result queryTaskListPaging(@ApiIgnore @RequestAttribute(value = Constants
...
                                       @RequestParam("pageSize") Integer pageSize) {
 
-        logger.info("query task instance list, project name:{},process instance:{}, search value:{},task name:{}, executor name: {},state type:{}, host:{}, start:{}, end:{}",
-                projectName, processInstanceId, searchVal, taskName, executorName, stateType, host, startTime, endTime);
+        logger.info("query task instance list, projectName:{}, processInstanceId:{}, processInstanceName:{}, search value:{}, taskName:{}, executorName: {}, stateType:{}, host:{}, start:{}, end:{}",
+                StringUtils.replaceNRTtoUnderline(projectName),
+                processInstanceId,
+                StringUtils.replaceNRTtoUnderline(processInstanceName),
+                StringUtils.replaceNRTtoUnderline(searchVal),
+                StringUtils.replaceNRTtoUnderline(taskName),
```
Comment (felix-thinkingdata) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TaskInstanceController.java:118]: if taskName is null ,will throw
 java.lang.NullPointerException: null

Comment (zhuangchong) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TaskInstanceController.java:118]: I add a null operation in StringUtils.replaceNRTtoUnderline

**Pull request merged**